client/extensions/woocommerce/app/orders/orders-list.js

```diff
@@ -5,45 +5,55 @@ import { bindActionCreators } from 'redux';
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
 import React, { Component } from 'react';
-import { times } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import Button from 'components/button';
 import EmptyContent from 'components/empty-content';
 import { fetchOrders } from 'woocommerce/state/sites/orders/actions';
 import formatCurrency from 'lib/format-currency';
 import {
 	areOrdersLoading,
 	areOrdersLoaded,
 	getOrders,
-	getTotalOrdersPages
+	getTotalOrders
 } from 'woocommerce/state/sites/orders/selectors';
 import { getLink } from 'woocommerce/lib/nav-utils';
-import { getOrdersCurrentPage } from 'woocommerce/state/ui/orders/selectors';
+import { getOrdersCurrentPage, getOrdersCurrentStatus } from 'woocommerce/state/ui/orders/selectors';
 import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
 import humanDate from 'lib/human-date';
-import { setCurrentPage } from 'woocommerce/state/ui/orders/actions';
+import { setCurrentQuery } from 'woocommerce/state/ui/orders/actions';
 import NavItem from 'components/section-nav/item';
 import NavTabs from 'components/section-nav/tabs';
+import Pagination from 'my-sites/stats/pagination';
 import SectionNav from 'components/section-nav';
 import Table from 'woocommerce/components/table';
 import TableRow from 'woocommerce/components/table/table-row';
 import TableItem from 'woocommerce/components/table/table-item';
 
 class Orders extends Component {
 	componentDidMount() {
-		const { siteId, currentPage } = this.props;
-
+		const { siteId, currentPage, currentStatus } = this.props;
+		const query = {
+			page: currentPage,
+			status: currentStatus,
+		};
 		if ( siteId ) {
-			this.props.fetchOrders( siteId, currentPage );
+			this.props.fetchOrders( siteId, query );
 		}
 	}
 
 	componentWillReceiveProps( newProps ) {
-		if ( newProps.currentPage !== this.props.currentPage || newProps.siteId !== this.props.siteId ) {
-			this.props.fetchOrders( newProps.siteId, newProps.currentPage );
+		if (
```

Comment:
Is it at all possible for newProps.siteId to be null? And if so, we probably need to avoid dispatching fetchOrders then.